build_and_upload.py

```
@@ -0,0 +1,54 @@
+import subprocess
+import requests
+import sys
+import os
+
+IP = "192.168.1.133"
+ENV = "xiao_esp32c3_wled_fix"
+PIO = os.path.expanduser("~\\.platformio\\penv\\Scripts\\platformio.exe")
+
+def get_bin_path():
+    result = subprocess.run(
+        ["git", "rev-parse", "--abbrev-ref", "HEAD"],
```

Ruff (0.15.17)

[error] Starting a process with a partial executable path

(S607)